### ISSUE TITLE
Unable to Install Native Messenger

### ISSUE THREAD

**Issue by gcmurray1990** [tridactyl/tridactyl#3753]
Hello, I am having some trouble installing the native messenger. 
I ran "curl -fsSl https://raw.githubusercontent.com/tridactyl/native_messenger/master/installers/install.sh -o /tmp/trinativeinstall.sh && sh /tmp/trinativeinstall.sh 1.21.1" in the command line and got this message:
**sh: 0: Can't open /tmp/trinativeinstall.sh**

after seeing this error message, I took the following steps as an attempted workaround:
1. Downloaded the native messenger repo from https://github.com/tridactyl/native_messenger
2. In the command line, cd to installers in the downloaded native messenger repo.
3. Run "sudo bash install.sh"
This is what I get from the command above:
**curl: (23) Failure writing output to destination
curl: (23) Failure writing output to destination
Failed to create '/root/.mozilla/native-messaging-hosts//tridactyl.json'. Please make sure that the directories exist and that you have the necessary permissions.
Installing manifest here: /root/.mozilla/native-messaging-hosts/
Installing script here: XDG_DATA_HOME: /root/.local/share/tridactyl
Failed to install!**

What am I doing wrong? 

Tridactyl version: 1.21.1
Firefox version: Mozilla Firefox 89.0
Operating system: linux(ubuntu 20.04)

**Comment by Rummskartoffel** [comment]
I don't know why you're getting an error accessing `/tmp/trinativeinstall.sh`, but trying it as root isn't going to help. Try navigating to `<repo>/installers` again and just run `sh ./install.sh 1.21.1`.

**Comment by gcmurray1990** [comment]
Navigating to /installers and running "sh ./install.sh 1.21.1" gives me the same output essentially:
**curl: (23) Failure writing output to destination
Installing manifest here: /home/gavin/.mozilla/native-messaging-hosts/
Installing script here: XDG_DATA_HOME: /home/gavin/.local/share/tridactyl
Failed to install!**

**Comment by Rummskartoffel** [comment]
In that case, sorry, no idea. I don't even know how you're getting an error message but still reaching the two "Installing [...]" messages  – because the script is supposed to exit immediately when we hit an error  –, let alone what the actual error might be. A quick Google suggests this error message can occur when you're piping somewhere, but we're not doing that if I'm not sorely mistaken.
Maybe @bovine3dom has some more ideas?

**Comment by gcmurray1990** [comment]
Piping occurs on line 97 of install.sh
`native_version="$(curl -sSL https://api.github.com/repos/tridactyl/native_messenger/releases/latest | grep "tag_name" | cut -d':' -f2- | sed 's|[^0-9\.]||g')"`

According to https://stackoverflow.com/questions/16703647/why-does-curl-return-error-23-failed-writing-body, changed that line t
`native_version="$(curl -sSL https://api.github.com/repos/tridactyl/native_messenger/releases/latest |tac|tac| grep "tag_name" | cut -d':' -f2- | sed 's|[^0-9\.]||g')"`

After that, I reran "sh ./install.sh 1.21.1" and got the same error.

**Comment by bovine3dom** [comment]
I don't have any good ideas other than maybe running the script bit by bit by hand to work out where the error is. It's a fairly simple script really.

What does `curl -sSL https://api.github.com/repos/tridactyl/native_messenger/releases/latest | grep "tag_name" ` do, for example?

**Comment by Rummskartoffel** [comment]
> Piping occurs on line 49 of install.sh

Yes, but when you invoke the script with a version number (<code>sh ./install.sh <em>1.21.1</em></code>) you're not even supposed to hit that line.

**Comment by gcmurray1990** [comment]
`Yes, but when you invoke the script with a version number (sh ./install.sh 1.21.1) you're not even supposed to hit that line.`
You're right, my bad.
Running line 47:`curl -sSL https://raw.githubusercontent.com/tridactyl/tridactyl/1.21.1/native/current_native_version 2>/dev/null` returns "0.3.4". So this doesn't seem to be the problem.

Line 65 and 66 is where the error is. Running line 65 without the silent flag (`curl -SL --create-dirs -o "/home/gavin/.mozilla/native-messaging-hosts/tridactyl.json" "https://raw.githubusercontent.com/tridactyl/native_messenger/0.3.4/tridactyl.json"`
) returns 
Warning: Failed to create the file 
Warning: /home/gavin/.mozilla/native-messaging-hosts/tridactyl.json: 
Warning: Permission denied
Permissions for /native-messaging-hosts are drwxr-xr-x. So not sure why I wouldn't have permission to write there.

**Comment by Rummskartoffel** [comment]
What happens when you manually `touch` the file? What happens when you try the command again, making sure that no instance of Firefox is running (you usually only need to do that on Windows, but it can't hurt to try)?

**Comment by gcmurray1990** [comment]
After running `touch /home/gavin/.mozilla/native-messaging-hosts/tridactyl.json` making sure Firefox is closed, and then rerunning `curl -SL --create-dirs -o "/home/gavin/.mozilla/native-messaging-hosts/tridactyl.json" "https://raw.githubusercontent.com/tridactyl/native_messenger/0.3.4/tridactyl.json `, I get the same exact error again unfortunately. 

On the bright side, I was able to get the installer to work by changing lines 65 and 66 to ```
curl -sSL   "$manifest_loc"|tee "$manifest_file"
    curl -sSL   "$native_loc"|tee "$native_file"
```
Thanks for helping me look into this.
